tests/transforms/test_defer_measurements.py

```diff
@@ -646,6 +649,172 @@ def cond_qnode(x, y):
         assert np.allclose(qml.matrix(normal_circuit)(x, y), qml.matrix(cond_qnode)(x, y))
 
 
+class TestExpressionConditionals:
+    """Test Conditionals that rely on expressions of mid-circuit measurements."""
+
+    @pytest.mark.parametrize("r", np.linspace(0.1, 2 * np.pi - 0.1, 4))
+    @pytest.mark.parametrize("op", [qml.RX, qml.RY, qml.RZ])
+    def test_conditional_rotations(self, r, op):
+        """Test that the quantum conditional operations match the output of
+        controlled rotations. And additionally that summing measurements works as expected."""
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def normal_circuit(rads):
+            qml.RX(2.4, wires=0)
+            qml.RY(1.3, wires=1)
+            qml.ctrl(op, control=(0, 1), control_values=[True, True])(rads, wires=2)
+            return qml.probs(wires=2)
+
+        @qml.qnode(dev)
+        @qml.defer_measurements
+        def quantum_control_circuit(rads):
+            qml.RX(2.4, wires=0)
+            qml.RY(1.3, wires=1)
+            m_0 = qml.measure(0)
+            m_1 = qml.measure(1)
+            qml.cond(m_0 + m_1 == 2, op)(rads, wires=2)
+            return qml.probs(wires=2)
+
+        normal_probs = normal_circuit(r)
+        cond_probs = quantum_control_circuit(r)
+
+        assert np.allclose(normal_probs, cond_probs)
+
+    @pytest.mark.parametrize("r", np.linspace(0.1, 2 * np.pi - 0.1, 4))
+    def test_triple_measurement_condition_expression(self, r):
+        """test that combining the results of three mid-circuit measurements works as expected."""
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        @qml.defer_measurements
+        def normal_circuit(rads):
+            qml.RX(2.4, wires=0)
+            qml.RY(1.3, wires=1)
+            qml.RX(1.7, wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[0, 2])
+
+            qml.ctrl(qml.RX, (0, 1, 2), [False, True, True])(rads, wires=3)
+
+            return qml.probs(wires=3)
+
+        @qml.qnode(dev)
+        @qml.defer_measurements
+        def quantum_control_circuit(rads):
+            qml.RX(2.4, wires=0)
+            qml.RY(1.3, wires=1)
+            qml.RX(1.7, wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[0, 2])
+            m_0 = qml.measure(0)
+            m_1 = qml.measure(1)
+            m_2 = qml.measure(2)
+
+            expression = 4 * m_0 + 2 * m_1 + m_2
+            qml.cond(expression == 3, qml.RX)(rads, wires=3)
+            return qml.probs(wires=3)
+
+        normal_probs = normal_circuit(r)
+        cond_probs = quantum_control_circuit(r)
+
+        assert np.allclose(normal_probs, cond_probs)
+
+    def test_multiple_conditions(self):
```

Comment:
As mentioned before, I think adding 2-3 different parameters to this and following test cases could be beneficial via parametrization.